### PR TITLE
[#9936] Add logging and swap if condition order

### PR DESCRIPTION
**Pull request by xpdavid**
Part of #9936

**This is not a fix for #9936**, we just add more logging to see what had happened and move the expensive if-condition to the end.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -71,7 +71,7 @@ dependencies {
     annotationProcessor objectify
 
     compile         "com.google.appengine:appengine-api-1.0-sdk:${appengineVersion}",
-                    "com.google.appengine.tools:appengine-gcs-client:0.7",
+                    "com.google.appengine.tools:appengine-gcs-client:0.8",
```
Comment (Adoby7) [inline build.gradle]: Is it necessary to update package in this PR, or should we move it to a separate PR?

Comment (xpdavid) [reply build.gradle]: Somehow the dependency of the package is missing in the maven center: https://travis-ci.org/TEAMMATES/teammates/builds/644947137?utm_source=github_status&utm_medium=notification

The only why to make it build is to upgrade to a new version.

Comment (Adoby7) [reply build.gradle]: OK, I think I will approve this PR after the travis issue being solved

**Comment by xpdavid** [comment]
The fix has been deployed. No error occurred so far.
